projects/plugins/jetpack/modules/widget-visibility/editor/index.jsx

```diff
@@ -0,0 +1,385 @@
+ *    { value: "Post type:", label: "Post type:", disabled: true },
+ *    { value: "post_type-post", label: "   Post" },
+ *    { value: "post_type-page", label: "   Page" },
+ *    { value: "post_type-attachment", label: "   Media" },
+ * ]
+ */
+
+const buildOptions = ( options, level = 0 ) =>
+	options.reduce( ( acc, item ) => {
+		const [ item1, item2 ] = item;
+		const prefix = '\u00A0'.repeat( level * 3 );
+
+		if ( Array.isArray( item2 ) ) {
+			const newItem = {
+				label: prefix + item1,
+				value: item1,
+				disabled: true,
+			};
+			const childItems = buildOptions( item2, level + 1 );
+			return acc.concat( [ newItem ] ).concat( childItems );
+		}
+
+		const newItem = {
+			label: prefix + item2,
+			value: item1,
+		};
+		return acc.concat( [ newItem ] );
+	}, [] );
+
+const VisibilityRule = props => {
+	const { rule, onDelete, setMajor, setMinor } = props;
+
+	// "User" and "Role" are hidden on wpcom
+	const optionsDisabledOnWpcom = [
+		{ label: __( 'User', 'jetpack' ), value: 'loggedin' },
+		{ label: __( 'Role', 'jetpack' ), value: 'role' },
+	];
+	const isWpcom = typeof wpcom !== 'undefined';
+
+	// "Taxonomy" is shown if there is at least one taxonomy (or if the current
+	// rule is taxonomy, so they can delete an invalid rule after removing
+	// taxonomies)
+	const isShowTaxonomy =
+		( widget_conditions_data.taxonomy && widget_conditions_data.taxonomy.length > 1 ) ||
+		rule.major === 'taxonomy';
+	const optionTaxonomy = [ { label: __( 'Taxonomy', 'jetpack' ), value: 'taxonomy' } ];
+
+	// Build options, but include user/role/taxonomy as appropriate
+	const majorOptions = [
+		{ label: __( '-- Select --', 'jetpack' ), value: '' },
+		{ label: __( 'Category', 'jetpack' ), value: 'category' },
+		{ label: __( 'Author', 'jetpack' ), value: 'author' },
+	]
+		.concat( isWpcom ? [] : optionsDisabledOnWpcom )
+		.concat( [
+			{ label: __( 'Tag', 'jetpack' ), value: 'tag' },
+			{ label: __( 'Date', 'jetpack' ), value: 'date' },
+			{ label: __( 'Page', 'jetpack' ), value: 'page' },
+		] )
+		.concat( isShowTaxonomy ? optionTaxonomy : [] );
+
+	let minorOptions = [];
+	if ( rule.major in widget_conditions_data ) {
+		minorOptions = buildOptions( widget_conditions_data[ rule.major ] );
+	}
+
+	return (
+		<div className="widget-vis__rule widget-vis__flex">
+			<div className="widget-vis__rule-col-1">
+				<div className="widget-vis__flex">
+					<div className="widget-vis__delete-rule">
+						<Button onClick={ onDelete } label={ __( 'Delete this visibility rule', 'jetpack' ) }>
+							<Icon icon={ close } size={ 14 } />
+						</Button>
+					</div>
+					<div className="widget-vis__select">
+						<SelectControl
+							label="Major Rule"
+							hideLabelFromVision
+							value={ rule.major }
+							options={ majorOptions }
+							onChange={ setMajor }
+						/>
+					</div>
+				</div>
+			</div>
+			<div className="widget-vis__rule-col-2">
+				<p className="widget-vis__is">{ __( 'is', 'jetpack' ) }</p>
```

Comment:
I think we'll need context on this string, like on the PHP version:
https://github.com/Automattic/jetpack/blob/3f8e9de7051fe6b26bcbb861de409a7e5aae1cd1/projects/plugins/jetpack/modules/widget-visibility/widget-conditions.php#L494